tests/test_object.h

```diff
@@ -87,6 +87,97 @@ TEST_CASE("[Object] Metadata") {
 			meta_list2.size() == 0,
 			"The metadata list should contain 0 items after removing all metadata items.");
 }
+
+TEST_CASE("[Object] IAPI methods") {
+	Object object;
+    bool *rvalid = memnew(bool(false));
```

Comment:
I think this adds unnecessary complexity, I'd rewrite this to:
```suggestion
    bool valid = false;
```
and then pass the pointer to methods which use this variable, such as below:
```c++
object.get_indexed(p_names, &valid);
```
Also, this way you don't have to dereference the pointer (`*valid`) whenever you need to reset the `valid` variable for other assertions.